common/rpc/encryption/localStoreCertProvider.go

```diff
@@ -52,22 +53,36 @@ func (s *localStoreCertProvider) GetSettings() *config.GroupTLS {
 }
 
 func (s *localStoreCertProvider) FetchServerCertificate() (*tls.Certificate, error) {
-	if s.tlsSettings.Server.CertFile == "" {
+	if !s.tlsSettings.Server.InlineData && s.tlsSettings.Server.CertFile == "" {
+		return nil, nil
+	}
+
+	if s.tlsSettings.Server.InlineData && s.tlsSettings.Server.CertData == "" {
```

Comment:
Shouldn't we return an error here?

Reply:
removed the boolean flag, which negates need to return an error